proposals/evaluation/uc-3-inheritance.md

```diff
@@ -0,0 +1,84 @@
+# Access Inheritance Use Cases
+
+Within this document, the following namespace prefix bindings are used:
+
+| Prefix  | Namespace                         |
+| ------- | --------------------------------- |
+| acl     | http://www.w3.org/ns/auth/acl#    |
+| acp     | http://www.w3.org/ns/solid/acp#   |
+| ex      | https://example.com/              |
+| ldp     | http://www.w3.org/ns/ldp#         |
+| vcard   | http://www.w3.org/2006/vcard/ns#  |
+
+## Read access on a collection of resources
+
+See also: https://solid.github.io/authorization-panel/authorization-ucr/#inheritance-readonly
+
+Alice can read all resources in a collection.
+
+### ACP
+
+```turtle
+ex:Collection1
+  acp:accessControl ex:AccessControl1 ;
+  ldp:contains ex:Resource1 .
+
+ex:AccessControl1
+  acp:applyMembers ex:Policy1 .
+
+ex:Policy1
+  acp:anyOf ex:AgentMatcher1 ;
+  acp:allow acl:Read .
+
+ex:AgentMatcher1
+  acp:agent ex:Alice .
+```
+
+Note: Resources linked to a collection via `ldp:contains` will inherit policies linked to the collection's access control via `acp:applyMembers`. Collections linked to a collection via `ldp:contains` will pass the inherited policies through to their member resources.
+
+### WAC
+
+The Weekly status collection is an `ldp:BasicContainer`, which contains a number of `ldp:BasicContainers`, one for each weekly meeting. The advantage of having these as containers rather than plain resources is that any number of other documents can be added to the container too.
+
+```turtle
+# Resource: </weekly-status/>
+<.>
+  a ldp:BasicContainer ;
+  ldp:contains <2021-04-28/>, <2021-05-05/>, <2021-05-12/> .
+```
+
+The `</weekly-status/.acl>` resource is advertised as `</weekly-status/>`'s access control list via a `Link` header with a relationship type of `http://www.w3.org/ns/auth/acl#accessControl`.
```

Comment:
Tentative information (until the WAC ED/WD goes through): `rel="acl"` (very likely), `rel="https://www.w3.org/ns/iana/link-relations/relation"`(possible in addition to former, or inferred), `rel="http://www.w3.org/ns/auth/acl#accessControl"` (less likely).

Reply:
I have a question about that specifically. In order to be consistent on the protocol level, could we have a standard relationship type linking resources to their access control resource?

What I mean is:
1. Using the same relationship type no matter what the authorization system is on the other side
2. Advertising itself as its own Access control resource when relevant (an acl mandates control over itself)

Then the resource type could be also consistently advertised by servers (something like rel="type", http://www.w3.org/ns/auth/acl#AccessControlList).

Reply:
1. Yes, that'd be ideal. Any reason why we can't use either one of the three I listed above? At least the way the `acl` link relation in the Protocol is A acl B with no constraints on A or B. And, the link-relation URI can be inferred or used in place.

2. https://github.com/solid/specification/issues/191

Reply:
While solid spec could add requirement that either link relation is valid `acl` or `https://www.w3.org/ns/iana/link-relations/relation#acl`. In plain HTTP they are two different link relations. I still don't understand why we can't just use an IRI as link relation and don't introduce this anomaly of requiring to treat two different link relations as equivalent.

Reply:
Having a standard registered link relation name for access control resources makes sense to me and hence supporting both `acl` and `https://www.w3.org/ns/iana/link-relations/relation#acl` on the client sounds good.

I think the alternative would be to not use a registered link relation name and go with a property defined in the most relevant ontology.

As far as I understand, it is likely that we would succeed in [getting a standard relation name registered](https://github.com/protocol-registries/link-relations) (for example `acl`).

> Any reason why we can't use either one of the three I listed above?

We could maybe consider registering a standard IANA link relation:

| Relation name    | Description                                                                                    | Reference         |
| ---------------- | -------------------------------------------------------------- | --------------- |
| access-control  | Refers to a resource mandating access over the link's context. | Solid Protocol   |

Reasons for spelling out "access control" instead of using "acl" could include:
- The specificity seems more adequate, that is, a resource that is not strictly speaking an `acl` could mandate control over the current context
- The currently registered standard relationships are not acronyms and it seems to me that the clear and explicit spelling "access-control" for a link relation would have positive effects on adoption (potentially beyond Solid and generally for access control discovery)
- It would eliminate the conflation of (or clarify the difference between) access control discovery and access control system

Reply:
I agree that we may want something that is less tied to "Access Control Lists". Those are very good starting points for access control, but we are already going beyond ACLs, and a lot of people may come to the wrong conclusion that the system we are building is limited to lists. 
But perhaps that should be discussed in its own issue, as it will get lost otherwise.

Reply:
Currently `acl` is *not* one of link relations registered with IANA https://www.iana.org/assignments/link-relations/link-relations.xhtml

I think we should just use an IRI and don't try to register anything with IANA.

Reply:
Could you explain benefit of registering a string vs using full IRI? We know that we do want possibility of using IRI to make statements in RDF. There is no official mapping of link relation to IRIs and we would need to define one for the link relation we consider registering. In that case I really can't see benefit of registering that would justify introducing two different link relations that should be considered as equivalent. If implementation MUST support the IRI link relation anyways, adding second alternative as short string registered with IANA only puts extra requirement (IMO pure burden) since full IRI link already MUST be supported.

Reply:
@bblfish @elf-pavlik , as per Henry's suggestion, let's follow up on this discussion there: https://github.com/solid/authorization-panel/issues/228

Elf, would it be be ok to start the debate with the question can we have 1 standard link relation for all access control systems?

But to your question, I think the registration of a new name with IANA for such a wide concept as access control discovery sounds like a good idea. If anything, I don't think it is important that Solid clients would understand the corresponding IRI as a Link relation name. On the other hand, for brevity, clarity and potential adoption, Solid servers using a Link rel="access-control" sounds good to me. And we can always use the IRI https://www.w3.org/ns/iana/link-relations/relation#access-control to express this relation in a resource representation (those statements that I consider part of the data-driven runtime of Solid servers).

However, I appreciate that those arguments are to some extent subjective.

That's why I would suggest starting with the other concerns.